soroban-env-host/src/budget/wasmi_helper.rs

```diff
@@ -141,3 +93,41 @@ impl ResourceLimiter for Host {
         WASMI_LIMITS_CONFIG.memories
     }
 }
+
+// These values are calibrated and set by us. Calibration is done with a given
+// wasmi version, and as long as the version is pinned, these values aren't
+// expected to change much.
+pub(crate) fn init_fuel_costs() -> FuelCosts {
+    let mut fuel_costs = FuelCosts::default();
+    fuel_costs.base = 1;
+    fuel_costs.entity = 3;
+    fuel_costs.load = 2;
+    fuel_costs.store = 1;
+    fuel_costs.call = 67;
+    fuel_costs
+}
+
+pub(crate) fn get_wasmi_config(host: &Host) -> Result<wasmi::Config, HostError> {
+    let protocol = host.get_ledger_protocol_version()?;
+    let mut config = wasmi::Config::default();
+    let fuel_costs = host.as_budget().0.try_borrow_or_err()?.fuel_costs;
+
+    // Turn off most optional wasm features, leaving on some
+    // post-MVP features commonly enabled by Rust and Clang.
+    config
+        .wasm_bulk_memory(true)
+        .wasm_multi_value(false)
+        .wasm_mutable_global(true)
+        .wasm_saturating_float_to_int(false)
+        .wasm_sign_extension(true)
+        .floats(false)
+        .consume_fuel(true)
+        .fuel_consumption_mode(FuelConsumptionMode::Eager)
+        .set_fuel_costs(fuel_costs);
+
+    if protocol > 20 {
+        config.wasm_reference_types(false);
```

Comment:
Is it possible to accidentally get this enabled in the build?

Reply:
what do you mean by that? This is a runtime config passed into vm instantiation

Reply:
Right, but does it invalidate certain Wasms? E.g. `floats` option invalidates Wasms that do floating point operations, right?

Reply:
The main purpose of the reference type proposal is to enable object passing between the host and the guest via a new reference type (`ExternRef`). It also expands the table operations and loosens some requirements such as allowing multiple tables. 
Our host design is not taking advantage of this feature by any means, and we are explicitly limiting (via the resource limiter) the number of tables to 1. So any legit contract built from the SDK should never touch this feature. The only way it can show up is if someone crafts it manually, which we want to disable them to reduce our maintenance surface area.